rlwe/rlwe_test.go

```diff
@@ -136,11 +136,42 @@ func testGenKeyPair(kgen KeyGenerator, t *testing.T) {
 
 	params := kgen.(*keyGenerator).params
 
+	sk, pk := kgen.GenKeyPair()
+
+	// Checks that the secret-key has exactly params.h non-zero coefficients
+	t.Run(testString(params, "SK"), func(t *testing.T) {
```

Comment:
Isn't this already tested in the package ring?

Reply:
Do you mean as a part of the `TernarySampler` tests ? 

We need to test it at that layer anyway, to catch bugs at the `KeyGen` level (e.g., using the wrong sampler).

Reply:
I'm agree